File: 351004/Beliaev/restapp/src/main/java/com/example/restapp/mapper/StickerMapper.java
package com.example.discussion.mapper;

import com.example.discussion.dto.request.StickerRequestTo;
import com.example.discussion.dto.response.StickerResponseTo;
import com.example.discussion.model.Sticker;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public interface StickerMapper {
    Sticker toEntity(StickerRequestTo request);
    StickerResponseTo toResponse(Sticker entity);

    @Mapping(target = "id", ignore = true)
    void updateEntityFromDto(StickerRequestTo request, @MappingTarget Sticker entity);
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/mapper/StickerMapper.kt
package com.example.distcomp.mapper

import com.example.distcomp.dto.request.StickerRequestTo
import com.example.distcomp.dto.response.StickerResponseTo
import com.example.distcomp.model.Sticker
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface StickerMapper {
    fun toEntity(request: StickerRequestTo): Sticker
    fun toResponse(entity: Sticker): StickerResponseTo
}
